suzieq/poller/worker/nodes/node.py

```diff
@@ -159,28 +159,30 @@ async def initialize(self, **kwargs) -> TNode:
 
         self._init_service_queue()
 
+        devtype = kwargs.get("devtype", None)
+        if devtype:
+            self._set_devtype(devtype, '')
+            self.logger.warning(
+                f'{devtype} supplied for {self.address}:{self.port}')
+
         if self.transport == "ssh":
             self.port = self.port or 22
             await self._init_ssh(init_dev_data=False)
         elif self.transport == "https":
             self.port = self.port or 443
-
-        devtype = kwargs.get("devtype", None)
-        if devtype:
-            self._set_devtype(devtype, '')
-            if self.transport == 'https':
+            if self.devtype:
                 await self._init_rest()
```

Comment:
Please add a comment here where you explain why we should init rest if the devtype is set. It's not clear to me

Reply:
OK. We need to add it because the devtype MAY not have been specified by user but transport provided is REST